dae/dae/backends/cnv/tests/test_configure_cnv_loader.py

```diff
@@ -0,0 +1,179 @@
+def test_configure_cnv_loader_vcf_like_pos():
+    header = ["person_id", "Chr", "Start", "End", "variant", "extra"]
+    transformers = []
+
+    _configure_cnv_location(
+        header, transformers,
+        cnv_chrom="Chr",
+        cnv_start="Start",
+        cnv_end="End"
+    )
+
+    assert header == [
+        "person_id", "chrom", "pos", "pos_end", "variant", "extra"]
+
+
+def test_configure_cnv_loader_vcf_like_pos_missmatch():
+
+    header = ["person_id", "Chr", "Start", "End", "variant", "extra"]
+    transformers = []
+
+    with pytest.raises(ValueError):
+        _configure_cnv_location(
+            header, transformers,
+            cnv_chrom="Chrom",
+            cnv_start="Start",
+            cnv_end="End"
+        )
+
+
+def test_configure_cnv_loader_location():
+    header = ["person_id", "location", "variant", "extra"]
+    transformers = []
+    _configure_cnv_location(header, transformers)
+    assert header == [
+        "person_id", "location", "variant", "extra"]
+
+
+def test_configure_cnv_loader_location_mismatch():
+    header = ["person_id", "location", "variant", "extra"]
+    transformers = []
+    with pytest.raises(ValueError):
+        _configure_cnv_location(
+            header, transformers, cnv_location="Location")
+
+
+def test_configure_cnv_loader_dae_best_state_default(
+        families, gpf_instance_2013):
+
+    header = ["family_id", "location", "variant", "best_state", "extra"]
+    transformers = []
+
+    (
+        header, transformers,
+        families, gpf_instance_2013.reference_genome,
+    )
+
+    assert header == [
+        "family_id", "location", "variant", "best_state", "extra"]
+
+
+def test_configure_cnv_loader_dae_best_state(
+        families, gpf_instance_2013):
+
+    header = ["familyId", "location", "variant", "bestState", "extra"]
+    transformers = []
+
+    _configure_cnv_best_state(
+        header, transformers,
+        families, gpf_instance_2013.reference_genome,
+        cnv_family_id="familyId",
+        cnv_best_state="bestState",
+    )
+
+    assert header == [
+        "family_id", "location", "variant", "best_state", "extra"]
+
+
+def test_configure_cnv_loader_person_id(
+        families, gpf_instance_2013):
+
+    header = ["personId", "location", "variant", "extra"]
+    transformers = []
+
+    _configure_cnv_best_state(
+        header, transformers,
+        families, gpf_instance_2013.reference_genome,
+        cnv_person_id="personId",
+    )
+
+    assert header == [
+        "person_id", "location", "variant", "extra"]
+
+
+def test_configure_cnv_loader_best_state_mixed(
+        families, gpf_instance_2013):
+
+    header = ["personId", "location", "variant", "bestState", "extra"]
+    transformers = []
+
+    with pytest.raises(ValueError):
+        _configure_cnv_best_state(
+            header, transformers,
+            families, gpf_instance_2013.reference_genome,
+            cnv_person_id="personId",
+            cnv_best_state="bestSTate",
```

Comment:
should this "bestSTate" string be misspelled?

Reply:
this test intentional error in the name of best_state... expects to raise ValueError